pika/data.py

```diff
@@ -121,67 +131,68 @@ def decode_value(encoded, offset):
     :raises: pika.exceptions.InvalidFieldTypeException
 
     """
-    kind = encoded[offset]
+    # slice to get bytes in Python 3 and str in Python 2
+    kind = encoded[offset:offset + 1]
```

Comment:
The `1-slice` here is to work around the difference that in Python 2 indexing a single character in `str` still yields `str`; in Python 3, indexing a single byte yields an integer, that is the value of the byte. But 1-slice of `bytes` is still `bytes` in Python 3